crates/eth_compatibility/src/lib.rs

```diff
@@ -0,0 +1,81 @@
+// Copyright 2018-2021 Parity Technologies (UK) Ltd.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#![cfg_attr(not(feature = "std"), no_std)]
+use ink_env::{
+    DefaultEnvironment,
+    Environment,
+};
+
+/// The ECDSA compressed public key.
+#[derive(Debug, Copy, Clone)]
+pub struct ECDSAPublicKey(pub [u8; 33]);
+
+impl Default for ECDSAPublicKey {
+    fn default() -> Self {
+        Self { 0: [0; 33] }
+    }
+}
+
+impl core::ops::Deref for ECDSAPublicKey {
+    type Target = [u8; 33];
+
+    #[inline]
+    fn deref(&self) -> &Self::Target {
+        &self.0
+    }
+}
+
+impl core::ops::DerefMut for ECDSAPublicKey {
+    #[inline]
+    fn deref_mut(&mut self) -> &mut Self::Target {
+        &mut self.0
+    }
+}
+
+/// Address of Ethereum account
+pub type EthereumAddress = [u8; 20];
```

Comment:
Could you make it a `pub struct EthereumAddress([u8; 20]);` with `Deref` and `DerefMut` implemented for it? So basically analog to what you did for `ECDSAPublicKey`.